### PR TITLE
Add E2E test using cypress

### PR DESCRIPTION
**Pull request by vidya-ram**

### CODE REVIEW

File: funnel/templates/project_cfp.html.jinja2 [hunk 1]
```diff
@@ -42,7 +42,7 @@
     {% endif %}
     <div class="mui-form form-actions clearfix">
       <div>
-        <button type="submit" name="submit" class="mui-btn mui-btn--raised mui-btn--primary" value="reject">Submit</button>
```
Comment (jace) [inline funnel/templates/project_cfp.html.jinja2]: This fix should be outside this PR.

File: funnel/templates/proposal.html.jinja2 [hunk 2]
```diff
@@ -107,6 +107,7 @@
               {%- endif %}
             {%- endfor %}
           {% endif %}
+          <span class="label mui--text-bold" data-cy-proposal-status="{{ proposal.state.label.title }}">Status: {{ proposal.state.label.title }}</span>
```
Comment (jace) [inline funnel/templates/proposal.html.jinja2]: Requires `{% trans %}`

File: funnel/templates/schedule.html.jinja2 [hunk 3]
```diff
@@ -108,7 +108,7 @@
...
                       <div class="schedule__row__column" :id="(view === 'agenda') ? 't' + key : ''" :class="['schedule__row__column--' + rowWidth, room.talk ? 'schedule__row__column--talks' : '', room.talk ? roomkey : '', room.talk && activeTab == roomkey ? 'js-active': '']" @click="showSessionModal(room.talk)"  :style="(getColumnWidth('content') > 0) ? {width: 'calc(100% - ' + getColumnWidth('content') + 'px)'} : {width: '100%'}" v-for="(room, roomkey, roomindex) in session.rooms">
                         <div class="schedule__row__column__content" :class="[view === 'calendar' ? 'schedule__row__column__content--calendar schedule__row__column__content--overflow' : '']" :style="(view == 'calendar') ? {height: getColumnHeight(room.talk.duration, rowHeight) + 'px'} : ''" :id="room.talk.url_name_suuid" v-if="room.talk">
-                          <p class="mui--text-body1 schedule__row__column__content__title__duration">{{ getTimeStr(room.talk.startTime) }}–{{ getTimeStr(room.talk.endTime) }}</p>
+                          <p class="mui--text-body1 schedule__row__column__content__title__duration">{{ getTimeStr(room.talk.startTime) }} - {{ getTimeStr(room.talk.endTime) }}</p>
```
Comment (jace) [inline funnel/templates/schedule.html.jinja2]: This should be an n-dash without spaces around it.

File: funnel/templates/session_view_popup.html.jinja2 [hunk 4]
```diff
@@ -18,7 +18,7 @@
       </p>
     {% endif %}
     {% if session.venue_room %}
-      <p class="mui--text-subhead zero-bottom-margin"><small class="mui--text-light">in</small> {{ session.venue_room.title }} <small class="mui--text-light">{{ session.venue_room.venue.title }}</small></p>
+      <p class="mui--text-subhead zero-bottom-margin" data-cy-session="room"><small class="mui--text-light">in {{ session.venue_room.venue.title }} – {{ session.venue_room.title }}</small></p>
```
Comment (jace) [inline funnel/templates/session_view_popup.html.jinja2]: UI change, should be separate from this PR.

File: funnel/templates/settings.html.jinja2 [hunk 5]
```diff
@@ -38,38 +38,38 @@
               {%- if project.current_roles.admin %}
                 {%- if transition_form %}
                   <div>
-                    <p class="mui--text-body2 zero-bottom-margin mui--text-uppercase">Status: {{ project.state.label.title }}</p>
+                    <p class="mui--text-body2 zero-bottom-margin mui--text-uppercase" data-cy="project-state">Status: {{ project.state.label.title }}</p>
```
Comment (jace) [inline funnel/templates/settings.html.jinja2]: `{% trans `%}` required.

File: funnel/templates/settings.html.jinja2 [hunk 6]
```diff
@@ -38,38 +38,38 @@
...
+                          <button name="transition" value="{{ name }}" class="transition mui-btn mui-btn--small mui-btn--raised {% if transition.data['type'] == 'success' %} mui-btn--primary {% elif transition.data['type'] == 'danger' %}  mui-btn--danger {%- endif %}" title="{{ transition.data['title'] }}"  data-cy-state="{{ name }}">{{ transition.data['title'] }}</button>
                         {% endfor %}
                       </p>
                     </form>
                   </div>
                 {%- endif %}
                 {%- if cfp_transition_form %}
                   <div>
-                    <p class="mui--text-body2 zero-bottom-margin mui--text-uppercase">{% trans %}CfP status{% endtrans %}: {{ project.cfp_state.label.title }}</p>
+                    <p class="mui--text-body2 zero-bottom-margin mui--text-uppercase" data-cy="cfp-state">{% trans %}CfP status{% endtrans %}: {{ project.cfp_state.label.title }}</p>
```
Comment (jace) [inline funnel/templates/settings.html.jinja2]: This is incorrect use of the `trans` tag. Correct usage:

```jinja2
<p ...>{% trans status=project.cfp_state.label.title %}CfP status: {{ status }}{% endtrans %}</p>
```

Please note that the entire block is translated, not just one fragment of text.

File: runfrontendtests.sh [hunk 7]
```diff
@@ -0,0 +1,8 @@
+#!/bin/sh
+set -e
+export PYTHONIOENCODING="UTF-8"
+export FLASK_ENV="TESTING"
+python -m tests.e2e.frontend_tests_initdb
+python runcypressserver.py &
```
Comment (jace) [inline runfrontendtests.sh]: This script doesn't kill the server after running tests. However, you can save the PID here and kill it afterwards. Change this to:

```sh
python runcypressserver.py &
SERVER_PID=$!
cd funnel/assets
npx cypress run --browser chrome --record --key $RECORD_KEY
kill $SERVER_PID
```

Two extra lines are inserted here.

File: funnel/assets/cypress/integration/10_updateSchedule.js [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+    cy.wait(60000);
+    cy.get('#select-date')
+      .type(tomorrow)
+      .click();
+    cy.get('.js-unscheduled').click();
+    cy.get('.fc-agenda-axis')
+      .contains(session.timecolumn)
+      .next('.fc-widget-content')
+      .click(5, 5);
+    cy.get('#popup').should('be.visible');
```
[the remark this answers is not in the document]
Comment (vidya-ram) [reply funnel/assets/cypress/integration/10_updateSchedule.js]: Wherever an element had an id attribute, that was used in the Cypress tests.
Incase of multiple elements of the same type, `data-cy-*` was added. 
Classes can get removed accidentally in the future PR's so added `data-cy-* `

File: runfrontendtests.sh (hunk 7 above)
Comment (vidya-ram) [reply runfrontendtests.sh]: Added these

File: funnel/templates/session_view_popup.html.jinja2 (hunk 4 above)
Comment (vidya-ram) [reply funnel/templates/session_view_popup.html.jinja2]: Changed it back

File: funnel/templates/schedule.html.jinja2 (hunk 3 above)
Comment (vidya-ram) [reply funnel/templates/schedule.html.jinja2]: Changed it back

File: funnel/assets/cypress/integration/08_rsvp.js [hunk 9]
```diff
@@ -0,0 +1,18 @@
+describe('RSVP yes to project', function() {
```
[the remark this answers is not in the document]
Comment (vidya-ram) [reply funnel/assets/cypress/integration/08_rsvp.js]: Changed

File: funnel/assets/cypress/integration/12_badge.js [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+
+    cy.relogin('/testcypressproject');
+    cy.get('a[data-cy-project="' + project.title + '"]').click();
+    cy.location('pathname').should('contain', project.url);
+    cy.get('a[data-cy-navbar="settings"]').click();
+    cy.location('pathname').should('contain', 'settings');
+    cy.get('a[data-cy="checkin"').click();
+    cy.location('pathname').should('contain', '/admin');
+    cy.get('a[data-cy="' + events[1].title + '"]').click();
+    cy.get('td[data-th="Name"]').contains(participants[2].fullname);
```
[the remark this answers is not in the document]
Comment (vidya-ram) [reply funnel/assets/cypress/integration/12_badge.js]: Changed to `data-cy`

File: funnel/assets/cypress/integration/16_viewSchedule.js [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+
+    cy.visit('/');
+    cy.get('.upcoming')
+      .find('.card--upcoming')
+      .contains(project.title)
+      .click({ force: true });
+    cy.get('[data-cy-navbar="schedule"]').click();
+    var tomorrow = Cypress.moment()
+      .add(1, 'days')
+      .format('dddd, D MMMM YYYY');
```
Comment (jace) [inline funnel/assets/cypress/integration/16_viewSchedule.js]: Isn't this hardcoding for locale?

**Pull request merged**